doc/SAI-IPv6-Segment-Routing-Update.md

```diff
@@ -0,0 +1,483 @@
+            CONVERT_STR_TO_IPV6(sidlist_entry_attrs[1].value.objlist.list[1], "2001:db8:85a3::8a2e:370:2345");
+            CONVERT_STR_TO_IPV6(sidlist_entry_attrs[1].value.objlist.list[2], "2001:db8:85a3::8a2e:370:3456");
+            saistatus = sai_v6sr_api->create_segmentroute_sidlist(&sidlist_id, switch_id, 2, sidlist_entry_attrs);    
+            
+        2. Create a tunnel object with source IP used for tunnel encapsulation
+
+            tunnel_entry_attrs[0].id = SAI_TUNNEL_ATTR_TYPE
+            tunnel_entry_attrs[0].value = SAI_TUNNEL_TYPE_SRV6
+            tunnel_entry_attrs[1].id = SAI_TUNNEL_ATTR_ENCAP_SRC_IP
+            CONVERT_STR_TO_IPV6(tunnel_entry_attrs[1].value, "2001:db8:85a3::8a2e:370:9876");
+            tunnel_entry_attrs[2].id = SAI_TUNNEL_ATTR_UNDERLAY_INTERFACE
+            tunnel_entry_attrs[2].value.oid = underlay_rif // created elsewhere
+            saistatus = sai_v6sr_api->create_tunnel(&tunnel_id, switch_id, 3, tunnel_entry_attrs)
+
+        3. Create a srv6 nexthop object bound to the SID list object and H.Encaps.Red behavior
+
+            nexthop_entry_attrs[0].id = SAI_NEXTHOP_ATTR_TYPE
+            nexthop_entry_attrs[0].value = SAI_NEXT_HOP_TYPE_SEGMENTROUTE_SIDLIST
+            nexthop_entry_attrs[1].id = SAI_NEXTHOP_ATTR_TUNNEL_ID
+            nexthop_entry_attrs[1].value.oid = tunnel_id
+            nexthop_entry_attrs[2].id = SAI_NEXT_HOP_ATTR_SEGMENTROUTE_SIDLIST_ID
+            nexthop_entry_attrs[2].value.oid = sidlist_id
+            saistatus = sai_v6sr_api->create_nexthop(&nexthop_id, switch_id, 3, nexthop_entry_attrs)
+             
+        4. Create a route entry which points to the srv6 nexthop
+
+            route_entry.switch_id = 0
+            route_entry.vr_id = vr_id_1 // created elsewhere
+            route_entry.destination.addr_family = SAI_IP_ADDR_FAMILY_IPV4
+            route_entry.destination.addr.ip4 = "198.51.100.0"
+            route_entry.destination.addr.mask = "255.255.255.0"            
+
+            route_entry_attrs[0].id = SAI_ROUTE_ENTRY_ATTR_NEXT_HOP_ID;
+            route_entry_attrs[0].value.oid = nexthop_id; 
+            saisstatus = saiv6sr_api->create_route(&route_entry, 1, route_entry_attrs)
+    
+
+- SR Endpoint/Transit
+    
+        Example configuration for End.DT46 behavior
+
+        local_sid_entry.switch_id = 0
+        local_sid_entry.vr_id = vr_id_1 // underlay VRF
+        local_sid_entry.locator_len = 64
+        local_sid_entry.function_len = 8
+        CONVERT_STR_TO_IPV6(local_sid_entry.sid, "2001:db8:0:1::1000:0:0:0");
+    
+        local_sid_attr[0].id = SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE
+        local_sid_attr[0].value = SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT46
+        local_sid_attr[1].id = SAI_LOCAL_SID_ENTRY_ATTR_VRF
+        local_sid_attr[1].value.oid = vr_id_1001 // overlay vrf, created elsewhere
+        saistatus = saiv6sr_api->create_local_sid(&local_sid_entry, 2, local_sid_attr)
```

Comment:
is this exact match or we can have lpm/route type lookup for my-local sid ?

Reply:
Changed SID from ip6_t to ip_prefix_it based on discussions in SAI review meeting.

Reply:
This topic is discussed in another section as well. In SRv6 workgroup, we agreed that application will program lengths for Locator Node, Locator block, function, arg fields and SAI adapter will calculate the appropriate prefix length from that information. Application doesn't need to explicitly set the prefix length.